src/components/Calculator.jsx

```diff
@@ -1,67 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import calculate from '../logic/calculate';
 import { numbers, signs } from '../static/button';
 import SignButton from './Buttons/BasicSigns';
 import NumberButton from './Buttons/NumberButton';
 import CalcInput from './CalcInput';
 
 // eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { calculations: {} };
-    this.handleClick = this.handleClick.bind(this);
-  }
-
-  handleClick(e) {
-    const { calculations } = this.state;
+const Calculator = () => {
+  const [calcs, setCalcs] = useState({});
+  useEffect(() => {
+    setCalcs({
+      calculations: {
+        total: 0,
+        next: 0,
+        operation: '',
+      },
+    });
+  }, []);
```

Comment:
[OPTIONAL] I think you could save a couple of lines here by just setting the state to its initial state inside of the `useState` call. There's no need to set the state inside of the useEffect. In fact, I would rewrite all of this as:
 
```bash
const [total, setTotal] = useState(0)
const [next, setNext] = useState(null)
const [operation, setOperation] = useState(null)
```
I would do it like this to avoid destructuring a lot later, this way I could have all of the values separate from the beginning. Of course, you would need to pass the three of them as an object to calculate function and destructure the result.

Reply:
I appreciate your review so much thank you 
but for this point, I read before when you are creating a complex state it's better to use useEffect to add its initial value so I implemented it like that now I am confused so is it wrong what I read before or what

Reply:
I don't think that's considered complex enough as no calculations or tasks are required to set the initial values. Something complex for the state I would of think O(n) problems at least. But that's why I marked this as optional and it was just my opinion. I might be wrong.

I get your point though, useEfffect runs asynchronously so the state calculating the state would not block the render of the component.

Reply:
Okay got it 
thanks again for your great effort :smile: